frontend/components/ActivityDetails/InstallDetails/SoftwareInstallDetailsModal/SoftwareInstallDetailsModal.tsx

```diff
@@ -0,0 +1,328 @@
+import deviceUserAPI from "services/entities/device_user";
+
+import InventoryVersions from "pages/hosts/details/components/InventoryVersions";
+
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+import Icon from "components/Icon";
+import Textarea from "components/Textarea";
+import DataError from "components/DataError/DataError";
+import DeviceUserError from "components/DeviceUserError";
+import Spinner from "components/Spinner/Spinner";
+import RevealButton from "components/buttons/RevealButton";
+import CustomLink from "components/CustomLink";
+
+import {
+  INSTALL_DETAILS_STATUS_ICONS,
+  getInstallDetailsStatusPredicate,
+} from "../constants";
+
+const baseClass = "software-install-details-modal";
+
+export type IPackageInstallDetails = {
+  host_display_name?: string;
+  install_uuid?: string; // not actually optional
+  deviceAuthToken?: string;
+};
+
+// TODO - match AppInstallDetailsModal status to this, still accounting for MDM-specific cases
+// present there
+const StatusMessage = ({
+  result: {
+    host_display_name,
+    software_package,
+    software_title,
+    status,
+    updated_at,
+    created_at,
+  },
+  isDUP,
+  isInstalledByFleet,
+}: {
+  result: ISoftwareInstallResult;
+  isDUP: boolean;
+  isInstalledByFleet: boolean;
+}) => {
+  const { config } = useContext(AppContext);
+
+  const formattedHost = host_display_name ? (
+    <b>{host_display_name}</b>
+  ) : (
+    "the host"
+  );
+
+  const displayTimeStamp = ["failed_install", "installed"].includes(
+    status || ""
+  )
+    ? ` (${formatDistanceToNow(new Date(updated_at || created_at), {
+        includeSeconds: true,
+        addSuffix: true,
+      })})`
+    : "";
+
+  const renderContactOption = () => (
+    // config undefined in the DUP context, so omit the link
+    <>
+      {" "}
+      or{" "}
+      {config?.org_info.contact_url ? (
+        <CustomLink
+          url={config.org_info.contact_url}
+          text="contact your IT admin"
+          newTab
+        />
+      ) : (
+        "contact your IT admin"
+      )}
+    </>
+  );
+
+  const renderStatusCopy = () => {
+    if (isInstalledByFleet) {
+      const prefix = (
+        <>
+          Fleet {getInstallDetailsStatusPredicate(status)}{" "}
+          <b>{software_title}</b>
+        </>
+      );
+      let middle = null;
+      if (isDUP) {
+        if (status === "failed_install") {
+          middle = <>. You can retry{renderContactOption()}</>;
+        }
+      } else {
+        // host details page
+        middle = (
+          <>
+            {" "}
+            ({software_package}) on {formattedHost}
+            {status === "pending_install" ? " when it comes online" : ""}
+            {displayTimeStamp}
+          </>
+        );
+      }
+      return (
+        <span>
+          {prefix}
+          {middle}
+          {"."}
+        </span>
+      );
+    }
+    if (status === "installed") {
```

Comment:
I think:
```
    if (status === "installed") {
      return (
        <span>
          <b>{software_title}</b> is installed.
        </span>
      );
    }
    return (
      <DataError description="Bad data from server - software cannot be both not installed (failed/pending) and not installed by Fleet" />
    );
  };
```
should be
```
      return (
        <span>
          <b>{software_title}</b> is installed.
        </span>
      );
    }
  };
```

Because if its `!isInstalledByFleet` but detected, in the installed_versions, they can get to this modal and should see "Software is installed." copy

Please cmiiw!

Reply:
I made some updates to this logic, lmk if this is still an issue